pkg/nixpath/chunker/chunker_test.go

```diff
@@ -0,0 +1,74 @@
+package chunker_test
+
+import (
+	"bytes"
+	_ "embed"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/nix-community/go-nix/pkg/nixpath/chunker"
+	"github.com/stretchr/testify/assert"
+)
+
+//go:embed simple.go
+var testData []byte
+
+func TestChunkers(t *testing.T) {
+	fastCDCChunker, err := chunker.NewFastCDCChunker(bytes.NewReader(testData))
+	if err != nil {
+		panic(err)
+	}
+
+	chunkers := []struct {
+		Name    string
+		Chunker chunker.Chunker
+	}{
+		{
+			"Simple",
+			chunker.NewSimpleChunker(bytes.NewReader(testData)),
+		},
+		{
+			"FastCDC",
+			fastCDCChunker,
+		},
+	}
+
+	for _, chunker := range chunkers {
+		t.Run(chunker.Name, func(t *testing.T) {
+			// grab data out of the chunker.
+			// Ensure it matches testData.
+
+			var receivedData bytes.Buffer
+			offset := uint64(0)
+
+			for {
+				chunk, err := chunker.Chunker.Next()
+				if err != nil {
+					if errors.Is(err, io.EOF) {
+						break
+					}
+					assert.NoError(t, err, "no other error other than EOF is accepted")
+				}
+				// check the chunk itself looks sane
+				assert.True(t,
+					uint64(len(chunk.Data)) == chunk.Size,
```

Comment:
how about having chunk.Size() that returns len(chunk.Data), so the data could never by unsynchronized?

Reply:
I initially picked the interface provided by fastcdc, but you're right, it's redundant. I'll apply the suggestion.

Reply:
Updated.